go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -778,9 +778,10 @@ func (qre *QueryExecutor) getConn() (*connpool.PooledConn, error) {
 	start := time.Now()
 	conn, err := qre.tsv.qe.conns.Get(ctx, qre.setting)
 
+	qre.logStats.WaitingForConnection += time.Since(start)
```

Comment:
It's probably better to do this in a defer call right after `start` is initialized. That will be future-proof.

Reply:
https://github.com/vitessio/vitess/pull/15073/commits/9d73ae1bfb2156de7f17fd18067b3150681e6353